File: app/src/main/java/br/com/samueltobias/customercrud/asynctask/FetchCustomersTask.java
package br.com.samueltobias.customercrud.asynctask;

import android.os.AsyncTask;

import java.util.List;

import br.com.samueltobias.customercrud.database.dao.CustomerDao;
import br.com.samueltobias.customercrud.model.Customer;

public class FetchCustomersTask extends AsyncTask<Void, Void, List<Customer>> {
    private CustomerDao customerDao;
    private Callback<List<Customer>> callback;

    public FetchCustomersTask(CustomerDao customerDao, Callback<List<Customer>> callback) {
        this.customerDao = customerDao;
        this.callback = callback;
    }

    @Override
    protected List<Customer> doInBackground(Void... voids) {
        return customerDao.getAll();
    }

    @Override
    protected void onPostExecute(List<Customer> customers) {
        super.onPostExecute(customers);

        callback.onFinish(customers);
    }
}

File: app/src/main/java/br/com/samueltobias/customercrud/data/repository/asynctask/FetchCustomersTask.kt
package br.com.samueltobias.customercrud.data.repository.asynctask

import android.os.AsyncTask
import br.com.samueltobias.customercrud.data.database.dao.CustomerDao
import br.com.samueltobias.customercrud.domain.model.Customer

//class FetchCustomersTask(
//        private val customerDao: CustomerDao,
//        private val callback: Callback<List<Customer>>
//) : AsyncTask<Void?, Void?, List<Customer>>() {
//    override fun doInBackground(vararg voids: Void?): List<Customer> {
//        return customerDao.getAll()
//    }
//
//    override fun onPostExecute(customers: List<Customer>) {
//        super.onPostExecute(customers)
//        callback.onFinish(customers)
//    }
//
//}